### PR TITLE
[MIG] machine

### PR DESCRIPTION
**Pull request by oerp-odoo**
Added footil dependency to reuse generate_names function for name_get
methods.
Implemented toggle_active propagation from machine template to its
childs.
Updated LT translation.
Added computation methods (with inverse) to compute amounts for database
system users, databases and OS users.
Updated demo data to match with Odoo 12.0

Updated travis for Odoo 12.0

### CODE REVIEW

File: machine/i18n/lt.po [hunk 1]
```diff
@@ -120,74 +130,74 @@ msgstr "Nustatymai"
 
 #. module: machine
 #: model:ir.model,name:machine.model_res_partner
-#: model:ir.model.fields,field_description:machine.field_machine_instance_partner_contact_id
+#: model:ir.model.fields,field_description:machine.field_machine_instance__partner_contact_id
 msgid "Contact"
-msgstr "Kontaktas"
+msgstr "Kontaktai"
```
Comment (SButko) [inline machine/i18n/lt.po]:
```suggestion
msgstr "Kontaktas"
```

File: machine/i18n/lt.po [hunk 2]
```diff
@@ -459,32 +504,32 @@ msgstr "Žemas"
...
 msgid "Machine Database System Name"
-msgstr "Machine Database System Name"
+msgstr "Mašinos duomenų bazės sistemos pavadinimas"
 
 #. module: machine
-#: model:ir.model.fields,field_description:machine.field_machine_email_recipient_machine_email_id
+#: model:ir.model.fields,field_description:machine.field_machine_email_recipient__machine_email_id
 msgid "Machine Email"
-msgstr "Mašinos el. paštas"
+msgstr "Mašnos el. paštas"
```
Comment (SButko) [inline machine/i18n/lt.po]:
```suggestion
msgstr "Mašinos el. paštas"
```

File: machine/i18n/lt.po [hunk 3]
```diff
@@ -517,45 +562,45 @@ msgid "Machine Instance Change Log"
...
 #: model:ir.model,name:machine.model_machine_dbs_instance_user
 msgid "Machine Instance DBS Users"
-msgstr "Machine Instance DBS Users"
+msgstr "Mašinos instancijos DBS vartotojai"
 
 #. module: machine
 #: model:ir.model,name:machine.model_machine_dbs_instance
 msgid "Machine Instance Database System"
-msgstr "Machine Instance Database System"
+msgstr "mašinos instancijos duomenų bazės sistema"
```
Comment (SButko) [inline machine/i18n/lt.po]:
```suggestion
msgstr "Mašinos instancijos duomenų bazės sistema"
```

File: machine/i18n/lt.po [hunk 4]
```diff
@@ -589,37 +634,52 @@ msgstr "Mašinos šablonai"
...
-#: model:ir.model.fields,field_description:machine.field_machine_email_mail_template_id
+#: model:ir.model.fields,field_description:machine.field_machine_email__mail_template_id
 msgid "Mail Template"
-msgstr "E-pašto šablonas"
+msgstr "Pašto šablonas"
+
+#. module: machine
+#: model:ir.model.fields,field_description:machine.field_machine_instance__message_main_attachment_id
+msgid "Main Attachment"
+msgstr "Pagrindinis prisegtukas"
```
Comment (SButko) [inline machine/i18n/lt.po]: Shouldn't we translate 'attachment' as 'priedas'? http://www.vlkk.lt/konsultacijos/169-attachment-atacmentas-priedas

File: machine/i18n/lt.po [hunk 5]
```diff
@@ -48,68 +53,73 @@ msgstr "Aktyvus"
 msgid "At least one recipient must be specified."
 msgstr "Turi būti nurodytas bent vienas gavėjas."
 
+#. module: machine
+#: model:ir.model.fields,field_description:machine.field_machine_instance__message_attachment_count
+msgid "Attachment Count"
+msgstr "Prisegtukų skaičius"
```
Comment (SButko) [inline machine/i18n/lt.po]: Shouldn't we translate 'attachment' as 'priedas'? http://www.vlkk.lt/konsultacijos/169-attachment-atacmentas-priedas

File: machine/i18n/lt.po [hunk 6]
```diff
@@ -778,22 +866,22 @@ msgid "Send Emails to Machine Contacts"
...
+#: model:ir.model.fields,field_description:machine.field_machine_instance__amount_storage_capacity
 msgid "Storage Capacity (GB)"
 msgstr "Saugojimo talpa (GB)"
 
 #. module: machine
-#: model:ir.model.fields,field_description:machine.field_machine_email_sub_subject
+#: model:ir.model.fields,field_description:machine.field_machine_email__sub_subject
 msgid "Sub Subject"
-msgstr "Potemė"
+msgstr "Sub Tema"
```
Comment (SButko) [inline machine/i18n/lt.po]: Why not 'Potemė'?

File: machine/views/machine_instance_views.xml [hunk 7]
```diff
@@ -191,8 +188,15 @@
         <field name="act_window_id" ref="machine_instance_action"/>
     </record>
 
+    <!-- It is more convenient to show instances by default instead
```
Comment (SButko) [inline machine/views/machine_instance_views.xml]: I totally agree with this :) I even had this on my wishlist :)

File: machine/views/machine_instance_views.xml [hunk 8]
```diff
@@ -6,11 +6,14 @@
...
         <field name="arch" type="xml">
-            <!-- TODO: might be better to have archive button on both forms.
-            Though when archiving template, it should archive its instances
-            too. -->
             <form>
                 <sheet>
+                    <div class="oe_button_box" name="button_box">
+                        <button name="toggle_active"
+                            type="object" class="oe_stat_button" icon="fa-archive">
+                            <field name="active" widget="boolean_button" options="{&quot;terminology&quot;: &quot;archive&quot;}"/>
```
Comment (SButko) [inline machine/views/machine_instance_views.xml]: I suggest to add a confirmation message something like in projects (e.g. confirm="(Un)archiving a project automatically (un)archives its tasks. Do you want to proceed?").

File: machine/i18n/lt.po (hunk 4 above)
Comment (oerp-odoo) [reply machine/i18n/lt.po]: Another autotranslate crap

File: machine/i18n/lt.po (hunk 6 above)
Comment (oerp-odoo) [reply machine/i18n/lt.po]: I dont know:)

File: machine/views/machine_instance_views.xml (hunk 8 above)
Comment (oerp-odoo) [reply machine/views/machine_instance_views.xml]: Maybe, but its good enough. Its easy to toggle back. Not that it will make a change with no return:)

**Review by SButko — APPROVED** [review]

**Pull request closed without merging**